### PR TITLE
Update UI for collections Edit Relationships tab

### LINKED ISSUE #1377: Collections: Create relationships partial of the new/edit form
### Descriptive summary

Add code to implement the Relationships tab partial for the new/edit form views for collections.  This partial is shared by both the new and edit form views.

### Rationale

See model issue for rationale.

### Implementation

**Files:**
* Edit [app/views/hyrax/collections/_form.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/_form.html.erb)
* Create [app/views/hyrax/collections/_form_relationships.html.erb](https://github.com/samvera/hyrax/blob/master/app/views/hyrax/collections/_form_relationships.html.erb)

#### Expected user interactions

Add the tab to the form by editing _form.html.erb.

Create the UI view for managing sub-collections for a collection by adding partial _relationships.html.erb.

Include sections for showing...
- [x] This Collection in Collections - Member of collection
- [x] Other Collections in this Collection
  - [ ] Attach New Collection (goes out an creates a new collection, then adds the relationship when done -- not working clean for works b/c you can't come back to editing the work you are adding the sub-work to via this approach)
  - [ ] Select existing collection to be a member sub-collection

Exact text for the page is in the mockup below.

#### Validation

Any changes to parent or sub-collections must validate that the path of collections is not circular.  This collection cannot be its own ancestor or descendant.

#### SPECIAL NOTE on Admin Sets:

For the initial implementaiton, the Admin Set underlying code will not change.  But to a user, the Admin Set will appear to be a collection.

To make the illusion happen...
* Nothing to do since admin sets can not nest.

### Mockup

collection-new-edit-relationships

#### Issues related to UI work for nesting

* #1417 original UI epic for admin index
* #1509 admin index - listing of collections - includes description of action menu items
* #1512 process `Add collection to another collection` action + modal for selecting other collection  (This one has the brain dump list of things needed for nesting)
* #1377 relationship partial for edit form

#### Collections original epic issues (mainly organized by UI)

| Collections feature | Issue |
| ------------------------ | ----- |
| Routes                   | #1380 |
| Controller               | # |
| Model                    | #1383 |
| View - Index            | #1417 |
| View - Show - Admin      | #1427 |
| View - Show - Landing    | #1428 |
| View - New/Edit          | #1374 |
| Partial -  Description   | #1375 |
| Partial -  Branding      | #1376 |
| Partial -  Relationships | (this issue) |
| Partial -  Sharing       | #1378 |
| Partial -  Discovery     | #1379 |

### PR DESCRIPTION
**Pull request by adamjarling**
Fixes #1377 

This is a partial fix, as it does not handle the actual wiring up of the UI components.  It just lays them out in the view.

### CODE REVIEW

File: app/views/hyrax/dashboard/collections/_form.html.erb [hunk 1]
```diff
@@ -7,6 +7,9 @@
       <li>
         <a href="#branding" role="tab" data-toggle="tab"><%= t('.tabs.branding') %></a>
       </li>
+      <li>
+        <a href="#relationships" role="tab" data-toggle="tab"><%= t('.tabs.relationships') %></a>
+      </li>
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_form.html.erb]: The relationships tab needs to be wrapped with `<% if @collection.nestable? %>`, such that it doesn't display the tab when `nestable? false`.

File: app/views/hyrax/dashboard/collections/_modal_remove_from_collection.html.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div class="modal fade" id="collection-remove-from-collection" tabindex="-1" role="dialog">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <form class="delete-collection-form">
+        <div class="modal-body">
+          <p>Removing this collection will not remove it from the repository, only from this parent colleciton.  Are you sure you want to remove this collection?</p>
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_modal_remove_from_collection.html.erb]: Typo:  "only from this parent colleciton"
Should be: "only from this parent **collection**"

Comment (elrayle) [reply app/views/hyrax/dashboard/collections/_modal_remove_from_collection.html.erb]: Can you move the string definition to hyrax.en and use I18n here?

File: app/views/hyrax/dashboard/collections/_modal_remove_from_collection.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-default" data-dismiss="modal"><%= t('helpers.action.cancel') %></button>
+          <button type="button" class="btn btn-danger">Remove</button>
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_modal_remove_from_collection.html.erb]: Get button label from hyrax.en using I18n.

File: app/views/hyrax/dashboard/collections/_modal_remove_sub_collection.html.erb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<div class="modal fade" id="collection-remove-sub-collection" tabindex="-1" role="dialog">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <form class="delete-collection-form">
+        <div class="modal-body">
+          <p>Removing this sub-collection will not remove it from the repository, only from this parent colleciton.  Are you sure you want to remove this sub-collection?</p>
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_modal_remove_sub_collection.html.erb]: Typo: "only from this parent colleciton"
Should be: "only from this parent **collection**"

Can you move the string definition to hyrax.en and use I18n here?

File: app/views/hyrax/dashboard/collections/_modal_remove_sub_collection.html.erb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn btn-default" data-dismiss="modal"><%= t('helpers.action.cancel') %></button>
+          <button type="button" class="btn btn-danger">Remove</button>
```
Comment (elrayle) [inline app/views/hyrax/dashboard/collections/_modal_remove_sub_collection.html.erb]: Get button label from hyrax.en using I18n.

**Comment by adamjarling** [comment]
@elrayle All updates mentioned above are fixed and pushed.

**Pull request merged**